staging/src/k8s.io/apiserver/pkg/audit/request.go

```diff
@@ -88,6 +89,11 @@ func NewEventFromRequest(req *http.Request, level auditinternal.Level, attribs a
 		}
 	}
 
+	annotations, _ := genericapirequest.AuditAnnotationsFrom(req.Context())
+	for key, value := range annotations {
+		LogAnnotation(ev, key, value)
+	}
```

Comment:
Should we silently ignore unfortunately placed annotations added after the request is setup? Do we want to make those calls panic instead? Can we reorg this in a way that supports e.g. annotations in an outbound portion of a filter?

Reply:
See what you think of the diff I proposed in 49bd215.  It should make it impossible to use `AuditAnnotationsFrom` later in the stack.

Reply:
Ah, this is the same as jordans comment.